channeldb/migration13/migration.go

```diff
@@ -114,15 +134,12 @@ func MigrateMPP(tx kvdb.RwTx) error {
 		}
 
 		// Save attempt id for later use.
-		attemptID := attemptInfo[:8]
-
-		// Discard attempt id. It will become a bucket key in the new
-		// structure.
-		attemptInfo = attemptInfo[8:]
+		copy(attemptID[:], attemptInfo[:8])
 
-		// Append unknown (zero) attempt time.
-		var zero [8]byte
-		attemptInfo = append(attemptInfo, zero[:]...)
+		// Copy over the attempt info into a new slice, the last 8
+		// bytes will be zero to mark a zero attempt time.
+		newAttemptInfo := make([]byte, len(attemptInfo))
```

Comment:
I think we can use a buffer pool both here and above, assuming the `Put` method copies the bytes over.

Reply:
`Put` method doesn't copy the `value` bytes over, it does copy the keys over: https://github.com/etcd-io/bbolt/blob/master/node.go#L139